test/controllers/articles_controller_test.rb

```diff
@@ -3,28 +3,86 @@
 require "test_helper"
 
 # Controller tests for news articles
-class ArticleControllerTest < FunctionalTestCase
-  def test_create_article_get
+class ArticlesControllerTest < FunctionalTestCase
+  ############ test Actions that Display data (index, show, etc.)
+
+  def test_index
+    # Prove any user can see article index
+    get(:index_articles)
+    assert(:success)
+
+    # Prove privileged user get link to create an article
+    login(users(:article_writer).login)
+    get(:index_articles)
+    assert_select("a", text: :create_article_title.l)
```

Comment:
Replace comment with a message. The message provides information both when reading the code and if the test fails. (I don't regard this suggestion as part of controller normalization.)
```suggestion
    login(users(:article_writer).login)
    get(:index_articles)
    assert_select("a", text: :create_article_title.l, "Privileged user should have link to create and article")
```